charts/karma/templates/deployment.yaml

```diff
@@ -93,11 +93,16 @@ spec:
 {{ toYaml . | indent 8 }}
     {{- end }}
       volumes:
-      {{- if .Values.configMap.enabled }}
+      {{- if and (.Values.configMap.enabled) (not .Values.secretConfig.enabled) }}
       - name: karma-config
         configMap:
           name: {{ .Release.Name }}-config
       {{- end }}
+      {{- if and (.Values.secretConfig.enabled) (not .Values.configMap.enabled) }}
```

Comment:
You could forbid to have the two parameters enabled:
```yaml
{{- if and .Values.secretConfig.enabled .Values.configMap.enabled }}
{{- fail "plz don't" }}
{{- end }}
```

And then simplify those conditions.

Reply:
That's a really good suggestion, thx for this one, I never used `fail` in Helm templates so far :)

Reply:
Great!